### ISSUE TITLE
View + Side Panel Toolbar interactions

### ISSUE THREAD

**Issue by caseyflynn-google** [eclipse-theia/theia#6187]
The interactions between the Side Panel Widgets and the View Menu seem a little odd compared to other IDEs. Currently clicking a widget name in the view menu will remove it from the side panel instead of focusing the widget.

**Comment by vince-fugnitto** [comment]
@caseyflynn-google I agree it's not intuitive, what do you believe the behavior should be?

Perhaps we can do the following (using the `search-in-workspace` (siw) widget as an example):

| Use Case | Behavior |
|:---|:---|
| Clicking `View` > `Search` <br /> (when siw is **hidden**) | The `search-in-workspace` widget is revealed  |
| Clicking `View` > `Search` <br /> (when siw is **visible**) | The `search-in-workspace` widget remains revealed  |
| Clicking `View` > `Search` <br /> (when siw is **doesn't exist**) | The `search-in-workspace` widget is spawned and revealed  |
| Clicking the `Search` widget tab <br /> (when siw is **hidden**) | The `search-in-workspace` widget is revealed |
| Clicking the `Search` widget tab <br /> (when siw is **visible**) | The `search-in-workspace` widget is collapsed |
| Right-clicking the `Search` widget tab and selecting close <br /> (when siw is **visible**) | The `search-in-workspace` widget is closed |

**Comment by akosyakov** [comment]
Added to discuss on next dev meeting. 

@vince-fugnitto as far as I understood that only the behavior of opening a view from the main menu bar should be changed, i.e. reveal and focus instead of toggling.

**Comment by vince-fugnitto** [comment]
> Added to discuss on next dev meeting.
> 
> @vince-fugnitto as far as I understood that only the behavior of opening a view from the main menu bar should be changed, i.e. reveal and focus instead of toggling.

Yes, I believe so too. I just wanted to explicitly describe each use case for test purposes later :)

**Comment by akosyakov** [comment]
PRs are welcomed!
